pkg/reconcile/pipeline/handler/project/impl.go

```diff
@@ -70,6 +70,11 @@ func BindingsAsEnv(ctx pipeline.Context) {
 		return
 	}
 
+	if len(ctx.BindingItems()) == 0 {
+		stop(ctx, errors.New("no binding item to project"))
```

Comment:
`stop` means that we are not going to retry, but we should actually.

Reply:
good point, thanks